node/assembler/assembler_metrics.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package assembler
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	arma_types "github.com/hyperledger/fabric-x-orderer/common/types"
+	node_ledger "github.com/hyperledger/fabric-x-orderer/node/ledger"
+)
+
+type AssemblerMetrics struct {
+	assembler *Assembler
+	index     PrefetchIndexer
+	shardIds  []arma_types.ShardID
+	partyIds  []arma_types.PartyID
+	logger    arma_types.Logger
+	interval  time.Duration
+	stopChan  chan struct{}
+	stopOnce  sync.Once
+}
```

Comment:
The metrics object should contain counters, and is given to the components. 
The components increment metrics counters when there are events, like in the router.
The metrics object should not contain links to individual components.

That is, lets do an inversion of control here. This will also suit Prometheus once we integrate it.